MaxText/checkpointing.py

```diff
@@ -49,27 +51,59 @@ def create_orbax_checkpoint_manager(
       options=CheckpointManagerOptions(
           create=True,
           save_interval_steps=save_interval_steps
-      )
+      ),
+      metadata={'iter': False}
   )
   max_logging.log("Checkpoint manager created!")
   return mngr
 
+def create_orbax_checkpoint_manager_pygrain(
```

Comment:
I'm very worried about this level of duplication making MaxText harder to understand and use. I think we have to hold back on this change until we're ready to always recommend Grain based on just this added mental load for users.

(I do think we can land lots of things separately.)